tracer/test/Datadog.Trace.Tests/Util/SpanCharSplitterTests.cs

```diff
@@ -0,0 +1,103 @@
+// <copyright file="SpanCharSplitterTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.Util;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Util;
+
+public class SpanCharSplitterTests
+{
+    [Fact]
+    public void SplitSpan()
+    {
+        var input = "hello;world;;a";
+        var enumerator = input.SplitIntoSpans(';').GetEnumerator();
+
+        // hello
+        enumerator.MoveNext().Should().BeTrue();
+        enumerator.Current.Length.Should().Be(5);
+        enumerator.Current.StartIndex.Should().Be(0);
+        enumerator.Current.AsSpan().ToArray().Should().BeEquivalentTo("hello");
+
+        // world
+        enumerator.MoveNext().Should().BeTrue();
+        enumerator.Current.Length.Should().Be(5);
+        enumerator.Current.StartIndex.Should().Be(6);
+        enumerator.Current.AsSpan().ToArray().Should().BeEquivalentTo("world");
+
+        // empty
+        enumerator.MoveNext().Should().BeTrue();
+        enumerator.Current.Length.Should().Be(0);
+        enumerator.Current.StartIndex.Should().Be(12);
+        enumerator.Current.AsSpan().ToArray().Should().BeEquivalentTo(string.Empty);
+
+        // a
+        enumerator.MoveNext().Should().BeTrue();
+        enumerator.Current.Length.Should().Be(1);
+        enumerator.Current.StartIndex.Should().Be(13);
+        enumerator.Current.AsSpan().ToArray().Should().BeEquivalentTo("a");
+
+        // end
+        enumerator.MoveNext().Should().BeFalse();
+    }
+
+    [Fact]
+    public void Count()
+    {
+        var input = "hello;world;;a";
+        var enumerator = input.SplitIntoSpans(';', 3).GetEnumerator();
+
+        // hello
+        enumerator.MoveNext().Should().BeTrue();
+        enumerator.Current.Length.Should().Be(5);
+        enumerator.Current.StartIndex.Should().Be(0);
+        enumerator.Current.AsSpan().ToArray().Should().BeEquivalentTo("hello");
+
+        // world
+        enumerator.MoveNext().Should().BeTrue();
+        enumerator.Current.Length.Should().Be(5);
+        enumerator.Current.StartIndex.Should().Be(6);
+        enumerator.Current.AsSpan().ToArray().Should().BeEquivalentTo("world");
+
+        // ;a
+        enumerator.MoveNext().Should().BeTrue();
+        enumerator.Current.Length.Should().Be(2);
+        enumerator.Current.StartIndex.Should().Be(12);
+        enumerator.Current.AsSpan().ToArray().Should().BeEquivalentTo(";a");
+
+        // end
+        enumerator.MoveNext().Should().BeFalse();
+    }
+
+    [Fact]
+    public void NoSeparator()
+    {
+        var input = "hello world";
```

Comment:
nit: tests with edge cases of `null` and with no spearator or empty where we specifiy a count might be nice (though pretty sure it's correct)

Reply:
Added in https://github.com/DataDog/dd-trace-dotnet/pull/5301/commits/f1af3493621372ca76eb2671624d5a782ad679df